lib/src/scheduler.rs

```diff
@@ -12,6 +12,11 @@ where
     fn schedule(&mut self) -> (i32, VecDeque<NodeIndex>);
 }
 
+pub trait DAGSetSchedulerBase<T: ProcessorBase + Clone, S: DAGSchedulerBase<T>> {
+    fn new(dag_set: Vec<Graph<NodeData, i32>>, processor: T, scheduler: S) -> Self;
```

Comment:
The `scheduler` object should not be held by DAGSetSchedulerBase, only DynamicFederatedScheduler.

Reply:
Changed DynamicFederatedScheduler to retain the scheduler.
Changed DynamicFederatedScheduler to retain the scheduler.
As a result, the set of processors in schedule() is no longer needed and has been removed.